GDJS/Runtime/variable.js

```diff
@@ -27,25 +43,38 @@ gdjs.Variable = function(varData)
 
 			//Try to guess the type of the value, as GD has no way ( for now ) to specify
 			//the type of a variable.
-			var valueWhenConsideredAsNumber = parseFloat(initialValue, 10);
+			var valueWhenConsideredAsNumber;
+			if(typeof initialValue === "string") {
```

Comment:
initialValue is currently always a string. I will keep this additional logic as it seems to work but this seems cumbersome and convoluted. Can we remove this in another PR? Or heavily simplify this?

This PR is already too big so I'll have to go ahead, but this is the kind of changes that should be separated because it requires testing and discussion.

Reply:
We should create issues to remember all thoose fixes that should be made

Reply:
Is it ok if I revert this change to this logic until we figure out something in another PR?
For code related issues, I would rather go for a "TODO" - keeping issues only for bugs as possible. 
In the sense that an open issue should be a signal for something that is a problem (creating problems for the users), while a TODO in the code is an indication of something that should be improved (but not fixing it immediately won't have the same impact on users)

Reply:
I think this might break Type Script type checking as TS won't the typing is number | string. What you would have to do is set the typing of _str and _value to {number | string} to keep it TS compliant.

Reply:
This is because "value" was typed as "number | string", but currently GDevelop is always putting a string in it ;) 
If you type 0.12 in GDevelop, it's saved as "0.12". So we have a string.

I do think it's interesting to do as you did to support numbers and strings, but this is a new feature/improvement => let's do it later :)